src/main/java/dev/hephaestus/atmosfera/client/sound/modifiers/implementations/BossBarCondition.java

```diff
@@ -0,0 +1,55 @@
+package dev.hephaestus.atmosfera.client.sound.modifiers.implementations;
+
+import com.google.gson.JsonObject;
+import dev.hephaestus.atmosfera.client.sound.modifiers.AtmosphericSoundModifier;
+import dev.hephaestus.atmosfera.world.context.EnvironmentContext;
+import net.minecraft.client.gui.hud.ClientBossBar;
+import net.minecraft.entity.boss.BossBar;
+import net.minecraft.text.TranslatableText;
+import net.minecraft.world.World;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.regex.Pattern;
+
+public record BossBarCondition(String text, boolean use_regex) implements AtmosphericSoundModifier, AtmosphericSoundModifier.Factory {
+
+    @Override
+    public float getModifier(EnvironmentContext context) {
+        if(!(text instanceof String)) {
+            return 0;
+        }
+
+        Collection<ClientBossBar> bossBars = context.getBossBars();
+
+        if (!bossBars.isEmpty()) {
+            Iterator var1 = bossBars.iterator();
+            while(var1.hasNext()) {
+                BossBar bossBar = (BossBar) var1.next();
+                String value = bossBar.getName() instanceof TranslatableText translatable ? translatable.getKey() : bossBar.getName().asString();
+
+                if(!use_regex && value.contains(text)) {
+                    return 1;
+                } else if(use_regex && Pattern.matches(text, value)) {
+                    return 1;
+                }
```

Comment:
Once we have the `Collection<String>` mentioned below, we can shorten this method quite a bit:
```java
if (this.isRegex) {
  for (String string : context.getBossBars()) {
    if (Pattern.matches(this.text, string)) return 1;
  }
} else if (context.getBossBars().contains(this.text)) {
  return 1;
}

return 0;
```